src/source/image_source.js

```diff
@@ -145,7 +146,12 @@ class ImageSource extends Evented implements Source {
             if (err) {
                 this.fire(new ErrorEvent(err));
             } else if (image) {
-                this.image = browser.getImageData(image);
+                const {HTMLImageElement} = window;
+                if (image instanceof HTMLImageElement) {
+                    this.image = browser.getImageData(image);
```

Comment:
Not a blocker for this PR, but for this code path, should we investigate caching of canvas creation separately? We could reuse and retain a canvas once created until we need to use it for an image that is has larger dimension. So that once the memory is there we would try to reuse it. For example:
- Create image of size 1024x1024, create canvas of size 1024x1024
- Create image of size 512x512, reuse canvas of size 1024x1024
- Create image of size 2048x2048, drop canvas of size 1024x1024, create new canvas of size 2048x2048
...

One case where this would work great is where all images are of the same size. At the moment, we're always re-requesting new canvas even when sizes are matching.

Reply:
I have an implementation of this that seems to work well and looks like it's fairly performant.

<img width="1438" alt="Screen Shot 2022-03-04 at 2 33 00 PM" src="https://user-images.githubusercontent.com/4523080/156851349-f7209098-5483-4809-8036-f34484e7fff1.png">